tests/function-controller/testsuite/teststep/configure.go

```diff
@@ -122,53 +174,20 @@ func (f ConfigureFunction) Run() error {
 }
 
 func (f ConfigureFunction) OnError() error {
-	if err := f.apiRule.LogResource(); err != nil {
-		return errors.Wrap(err, "while getting apirule")
-	}
-
-	if err := f.svcInstance.LogResource(); err != nil {
-		return errors.Wrap(err, "while getting service instance")
-	}
-
-	if err := f.svcBinding.LogResource(); err != nil {
-		return errors.Wrap(err, "while getting service binding")
-	}
-
-	if err := f.svcBindingUsage.LogResource(); err != nil {
-		return errors.Wrap(err, "while getting service binding usage")
-	}
-
-	if err := f.subscription.LogResource(); err != nil {
-		return errors.Wrap(err, "while getting subscription")
+	for _, object := range f.createdObjects {
+		if err := object.LogResource(); err != nil {
+			return errors.Wrapf(err, "while getting %s", object.GetName())
```

Comment:
I wonder if `OnError` and `Cleanup` functions should return early once they encounter the first error. Wouldn't it be better to try iterating over all the objects anyway?